repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -255,14 +263,40 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                         allowClauseList.add(mapOf("bool", mapOf("must", mustMap)));
                     }
 
-                } else if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_PRODUCT)) {
+                } else if (policyActions.contains(ACCESS_READ_PERSONA_PRODUCT)) {
                     for (String asset : assets) {
                         //terms.add(asset);
                         List<Map<String, Object>> mustMap = new ArrayList<>();
                         mustMap.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, asset + "/*product/*")));
                         mustMap.add(mapOf("term", mapOf("__typeName.keyword", "DataProduct")));
                         allowClauseList.add(mapOf("bool", mapOf("must", mustMap)));
                     }
+                } else if (policyActions.contains(ACCESS_READ_PERSONA_AI_ASSET)) {
+                    List<String> resources = getPolicyResources(policy);
+                    List<String> typeResources = getFilteredPolicyResources(resources, RESOURCES_ENTITY_TYPE);
+                    if (CollectionUtils.isNotEmpty(typeResources)) {
+                        List<String> typeTerms = new ArrayList<>();
+                        List<Map<String, Object>> mustMap = new ArrayList<>();
+                        if (typeResources.contains(AI_APPLICATION)) {
+                            typeTerms.add(AI_APPLICATION);
+                        } 
+                        if (typeResources.contains(AI_MODEL)) {
+                            typeTerms.add(AI_MODEL);
+                        }
+                        if (CollectionUtils.isNotEmpty(typeTerms)) {
+                            mustMap.add(mapOf("terms", mapOf("__typeName.keyword", typeTerms)));
+                            allowClauseList.add(mapOf("bool", mapOf("must", mustMap)));
+                        }
+                    }
```

Comment:
Suggesting following for simplicity (helps maintaining code in a better way for future)

```
if (policyActions.contains(ACCESS_READ_PERSONA_AI_ASSET)) {
      allowClauseList.add(mapOf("terms", mapOf("__typeName.keyword", Arrays.asList(AI_APPLICATION, AI_MODEL))));
      terms.addAll(assets);
}
```

Please review